analytics/analytics.go

```diff
@@ -162,3 +184,28 @@ func (dc *dataCollection) collectionAgent() {
 		dc.sendData()
 	}
 }
+
+func (dc *dataCollection) reportHealthDueToError(failurePoint string) {
```

Comment:
Beside error do we want warning as well?

Reply:
> Beside error do we want warning as well?

there is no warning instance in the analytics

Reply:
We may need a better function name to replace "reportHealthDueToError"?
How about reportError, reportHealthAlert ...?

Reply:
sure, renamed the function name to reportHealthAlert